test/lib/utils/swapTestDataHelpers.ts

```diff
@@ -0,0 +1,752 @@
+        decimalScale: tokenAmount.decimalScale.toString(),
+        scale18: tokenAmount.scale18.toString(),
+    };
+};
+
+/**
+ * Deserializes a TokenAmount from JSON.
+ * Validates the structure and data before creating the TokenAmount instance.
+ * @param serialized - The serialized TokenAmount data
+ * @returns A TokenAmount instance
+ * @throws Error if the data structure is invalid or validation fails
+ */
+export const deserializeTokenAmount = (serialized: unknown): TokenAmount => {
+    if (typeof serialized !== 'object' || serialized === null) {
+        throw new Error(
+            `Invalid serialized TokenAmount: expected object, got ${typeof serialized}`,
+        );
+    }
+
+    if (!('token' in serialized)) {
+        throw new Error(
+            'Invalid serialized TokenAmount: missing required field "token"',
+        );
+    }
+    if (!('amount' in serialized)) {
+        throw new Error(
+            'Invalid serialized TokenAmount: missing required field "amount"',
+        );
+    }
+
+    const data = serialized as {
+        token: {
+            chainId: number;
+            address: Address;
+            decimals: number;
+            symbol?: string;
+            name?: string;
+            wrapped?: Address;
+        };
+        amount: string;
+        scalar?: string;
+        decimalScale?: string;
+        scale18?: string;
+    };
+
+    // Validate token structure
+    if (
+        typeof data.token !== 'object' ||
+        data.token === null ||
+        typeof data.token.chainId !== 'number' ||
+        typeof data.token.address !== 'string' ||
+        typeof data.token.decimals !== 'number' ||
+        (data.token.wrapped !== undefined &&
+            typeof data.token.wrapped !== 'string')
+    ) {
+        throw new Error(
+            'Invalid serialized TokenAmount: token object is malformed or missing required fields (chainId, address, decimals)',
+        );
+    }
+
+    // Validate amount
+    if (typeof data.amount !== 'string') {
+        throw new Error(
+            `Invalid serialized TokenAmount: amount must be a string, got ${typeof data.amount}`,
+        );
+    }
+
+    let amountBigInt: bigint;
+    try {
+        amountBigInt = BigInt(data.amount);
+    } catch {
+        throw new Error(
+            `Invalid serialized TokenAmount: amount "${data.amount}" cannot be converted to BigInt`,
+        );
+    }
+
+    // Validate amount is non-negative
+    if (amountBigInt < 0n) {
+        throw new Error(
+            `Invalid serialized TokenAmount: amount cannot be negative, got ${data.amount}`,
+        );
+    }
+
+    // Validate address format (basic check - should be hex string starting with 0x)
+    if (!/^0x[a-fA-F0-9]{40}$/.test(data.token.address)) {
+        throw new Error(
+            `Invalid serialized TokenAmount: token address "${data.token.address}" is not a valid Ethereum address`,
+        );
+    }
+    if (
+        data.token.wrapped !== undefined &&
```

Comment:
This will likely need to be refactored after merging main back to this working branch since wrapped was removed from Token